Sources/Misc/SandboxEnvironmentDetector.swift

```diff
@@ -40,12 +44,19 @@ final class BundleSandboxEnvironmentDetector: SandboxEnvironmentDetector {
             return false
         }
 
-        // `true` for either `macOS` or `Catalyst`
-        let isMASReceipt = path.contains("MASReceipt/receipt")
-        if isMASReceipt {
-            return path.contains("Xcode/DerivedData")
-        } else {
+        #if os(macOS) || targetEnvironment(macCatalyst)
```

Comment:
Feel free to do it on a separate PR (maybe pointing to the macOS tests branch), but this could really use tests. I'm afraid that if we don't add them now we'll forget.